languages/ruby/lib/oso/polar/query.rb

```diff
@@ -118,9 +118,18 @@ def start # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metri
               id = event.data['instance_id']
               raise DuplicateInstanceRegistrationError, id if host.instance? id
 
-              cls_name = event.data['instance']['tag']
-              fields = event.data['instance']['fields']['fields']
-              host.make_instance(cls_name, fields: fields, id: id)
+              constructor = event.data['constructor']['value']
+              if constructor.key? 'InstanceLiteral'
+                cls_name = constructor['InstanceLiteral']['tag']
+                fields = constructor['InstanceLiteral']['fields']['fields']
+                initargs = Hash[fields.map { |k, v| [k.to_sym, host.to_ruby(v)] }]
+              elsif constructor.key? 'Call'
+                cls_name = constructor['Call']['name']
+                initargs = constructor['Call']['args'].map { |arg| host.to_ruby(arg) }
+              else
+                raise InvalidConstructorError
+              end
+              host.make_instance(cls_name, initargs: initargs, id: id)
```

Comment:
Could this be split into a separate function?

Reply:
I don't think that would buy anything. This logic is inherent to the task of instantiating an instance in one of several ways.

Reply:
Right, but then maybe it should go in `#make_instance` instead of in `#start`, and `#make_instance` gets called with the `constructor` JSON object?

Reply:
I'm sorry, my comment was poorly phrased. What I meant was that this is the decoding logic for the two cases we have now (boa and kwargs), and I like the strict separation we have now between decoding the messages (in `Query`) and implementing the host-specific bits (in `Host`). So I don't think it belongs in `Host.make_instance`.